### ISSUE TITLE
Event / Vaktsys

### ISSUE THREAD

**Issue by tellnes** [laafte/bull2#1]
Jeg ser for meg at styret får opp en kalender hvor de kan kjapt trykke nedover (alla checkboxer) hvilke datoer det skal være åpent. Utifra det vil åpningseventer automatisk gennereres. Metainformasjon skal kunne legges inn i etterkant.

**For styret:**
- [ ] Ny åpningsevent
  - [ ] Metainformasjon: Hva skjer den kvelden (temafest, konsert ol.)
  - [ ] Checkbox som den skal markedsføres som event
  - [ ] Gennerering av vakter
    - [ ] Barvakter (som standar to personer, mulighet til å endre antall)
    - [ ] Styrevakt(er) (To per dag under Låftefestivalen)
    - [ ] Vaskevakter dagen derpå
- [ ] Ikke medlemmer skal kunne ta vakter (legges inn av styret, i utgangspunktet ikke byttbar)
- [ ] Andre type vakter. Eksempler er:
  - Ølbæringsvakter
  - Dovask
  - Paukebæring
- [ ] Har vakten blitt gjennomført?
  - Hvis noen ikke møter til bar- eller vaskevakt, stryker vi en av Snau sine og gir det til gjengen som ikke har stilt med vakter
- [ ] Lenkevakter (går kanskje under "andre type vakter", men hvor kun styret kan ta dem)

Ikke alle typer vakter defineres som "du har gjort din jobb". F.eks. paukebæringsvakter gir poeng innad i Symforch, men det gir deg ikke Låfte-poeng.

**For medlemmer:**

Jeg tror dere har kontroll på denne biten selv, men legger inn et par punkter. En del av dette er nice to have funksjonalitet.
- [ ] Se liste over eventer (internt og ekstern)
  - Konserter med Låfte-grupperinger utenfor Låfte
  - Åpningsdager merket for markedsføring (SiGP, konserter ol)
- [ ] Fremside (el.) Hva skjer på Låfte ikveld/neste gang
- [ ] ical feed
- [ ] Merke en vakt som "bytting ønskes"
- [ ] Er det folk på Låfte nå?
  - [ ] Hva spilles på Spotify (last.fm)
  - [ ] Webcam?
  - [ ] Lydsensor?

**Comment by joharei** [comment]
Trenger vi å kunne endre tid/dato på individuelle ukentlige events (i hovedsak faste øvinger)? Det gjør nemlig implementasjonen litt mer komplisert, nok til at jeg vurderer å ta i bruk et rammeverk. Jeg vet ikke hvor ofte slike avvik oppstår, så inntil videre jobber jeg på et eget enkelt system.

**Comment by tellnes** [comment]
Hvis man ikke kan endre tidspunkt for en fast event, så bør styret kunne legge inn eventer som skjer samtidig. Egentlig kan man jo bare gjøre det slik at styret alltid kunne legge inn paralelle eventer, men med en advarsel.

Alle faste eventer burde også bekreftes av leder i den grupperingene de tilhører (styret i praksis). Faste eventer bør også kunne slettes. Det resulterer i at hvis noen ikke har øving en uke, så er det mulig for andre å booke den dagen. Jeg har selv ønsket å gjøre det noen ganger.

En enkel måte å impementere endring av fast arrangement på er å slette eventen den uken (et unntak) og legge det inn som en vanlig reservasjon. Oppføringen vil da bli uavhengig av den repetive eventen og ikke endres hvis den repetive endres.

**Comment by joharei** [comment]
Slik det er nå, lagres faste arrangementer som kun ett arrangement med en boolean `weekly`. Når kalenderviewet spør etter arrangementer for en uke, finner spørringen tidligere events med `weekly`. Hvis vi vil endre/slette individuelle events, kan man f.eks. lagre unntak istedet.

**Comment by tellnes** [comment]
Ja. Men du kan gjøre det hele sømløst i guiet ved å automatisk opprette untaket og opprette et enkelt aragement under en operasjon. Race conditionen som oppstår løses enkelt med transactions i databasen.

Jeg anbefaler også å eksperimentere litt med repetive hendelser i Google Calender. Funksjonaliteten der er faktisk ganse kraftig, men ikke overkomplisert å implementere.
